### PR TITLE
Make the reader get the Tables

### LINKED ISSUE #1158: Missing feature: reader doesn't retrieve the zcl_excel_table objects
Run the below program, which writes `The active sheet contains 0 tables.`:
```abap
TYPES tt_scarr TYPE STANDARD TABLE OF scarr WITH EMPTY KEY.
DATA(lt_scarr) = VALUE tt_scarr(
    ( carrid = 'AA' carrname = 'American Airlines' )
    ( carrid = 'LH' carrname = 'Lufthansa' ) ).

DATA(lo_excel) = NEW zcl_excel( ).
DATA(lo_worksheet) = lo_excel->get_active_worksheet( ).
lo_worksheet->bind_table( ip_table = lt_scarr ).
DATA(lo_writer) = CAST zif_excel_writer( NEW zcl_excel_writer_2007( ) ).
DATA(lv_xstring) = lo_writer->write_file( lo_excel ).

DATA(lo_excel_reader) = NEW zcl_excel_reader_2007( ).
DATA(lo_excel_2) = lo_excel_reader->zif_excel_reader~load( lv_xstring ).
DATA(lo_worksheet_2) = lo_excel_2->get_active_worksheet( ).

DATA(lv_table_size) = lo_worksheet_2->get_tables_size( ).

WRITE |The active sheet contains { lv_table_size } tables.|.
```
It should write `The active sheet contains 1 tables.`

Missing feature.

### PR DESCRIPTION
**Pull request by sandraros**
Fix #1158

### CODE REVIEW

**Comment by sandraros** [comment]
@larshp I don't get what all these abaplint messages are about. Is there anything I should do? Thanks.
image

File: src/zcl_excel_reader_2007.clas.abap [hunk 1]
```diff
@@ -3885,6 +3917,126 @@ CLASS zcl_excel_reader_2007 IMPLEMENTATION.
...
+    DATA: BEGIN OF ls_table_column,
+            id   TYPE string,
+            name TYPE string,
+          END OF ls_table_column.
+
+    FIELD-SYMBOLS:
+      <ls_table> LIKE LINE OF it_tables,
+      <lt_table> TYPE STANDARD TABLE.
+
+    LOOP AT it_tables ASSIGNING <ls_table>.
```
Comment (larshp) [inline src/zcl_excel_reader_2007.clas.abap:3962]: not 702 compatible

**Review by larshp — DISMISSED** [review]

File: src/zcl_excel_reader_2007.clas.abap [hunk 2]
```diff
@@ -3885,6 +3917,127 @@ CLASS zcl_excel_reader_2007 IMPLEMENTATION.
...
+        ls_field_catalog-dynpfld = abap_true.
+        ls_field_catalog-abap_type = cl_abap_typedescr=>typekind_string.
+        APPEND ls_field_catalog TO lt_field_catalog.
+
+        lo_ixml_table_column ?= lo_ixml_iterator->get_next( ).
+
+      ENDWHILE.
+
+      CLEAR lt_components.
+      LOOP AT lt_field_catalog ASSIGNING FIELD-SYMBOL(<field>).
```
Comment (larshp) [inline src/zcl_excel_reader_2007.clas.abap]: not 702 compatible

**Comment by sandraros** [comment]
@larshp Thanks for the support! :)

**Comment by sandraros** [comment]
Fix #1168

**Pull request merged**